scribble-lib/scribble/racket.rkt

```diff
@@ -793,50 +795,52 @@
                (set! src-col (+ src-col 1))))]
           [(box? (syntax-e c))
            (advance c init-line! srcless-step)
-           (let ([quote-depth (to-quoted c expr? quote-depth out color? inc-src-col)])
-             (if (and expr? (zero? quote-depth))
-                 (begin
-                   (out "(" paren-color)
-                   (out "box" symbol-color)
-                   (out " " #f)
-                   (set! src-col (+ src-col 5)))
-                 (begin
-                   (out "#&" value-color)
-                   (set! src-col (+ src-col 2))))
+           (let* ([quote-depth (to-quoted c expr? quote-depth out color? inc-src-col)]
+                  [expr-mode? (and expr? (zero? quote-depth))])
+             (cond
+               [expr-mode?
+                (out "(" paren-color)
+                (out "box" symbol-color)
+                (out " " #f)
+                (set! src-col (+ src-col 5))]
+               [else
+                (out "#&" value-color)
+                (set! src-col (+ src-col 2))])
              (hash-set! next-col-map src-col dest-col)
              ((loop init-line! (if expr? quote-depth +inf.0) expr? #f) (unbox (syntax-e c)) #f)
-             (when (and expr? (zero? quote-depth))
+             (when expr-mode?
                (out ")" paren-color)))]
           [(hash? (syntax-e c))
            (advance c init-line! srcless-step)
-           (let ([equal-table? (hash-equal? (syntax-e c))]
-                 [eqv-table? (hash-eqv? (syntax-e c))]
-                 [quote-depth (to-quoted c expr? quote-depth out color? inc-src-col)])
-             (unless (and expr? (zero? quote-depth))
+           (let* ([equal-table? (hash-equal? (syntax-e c))]
+                  [eqv-table? (hash-eqv? (syntax-e c))]
+                  [quote-depth (to-quoted c expr? quote-depth out color? inc-src-col)]
+                  [expr-mode? (and expr? (zero? quote-depth))])
+             (unless expr-mode?
                (out (if equal-table?
```

Comment:
@jackfirth why doesn't Resyntax convert this nested `if` to `cond`?

Reply:
It's not in the changed lines of the diff. Resyntax only comments on changed lines, since GitHub doesn't let reviews posted through the API comment on unchanged lines.